mmcv/runner/ipu/model_converter.py

```diff
@@ -0,0 +1,814 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+import poptorch
+import numpy as np
+import inspect
+import copy
+import warnings
+from typing import Optional, Union
+from collections import OrderedDict
+from poptorch import PoplarExecutor, __version__, identity_loss
+from poptorch._args_parser import ArgsParser
+from mmcv.parallel.data_container import DataContainer
+from ..fp16_utils import auto_fp16
+
+
+class DictArgsParser(ArgsParser):
+    """A helper class for handling model input.
+    Args:
+        inputs (list): Inputs of model.
+    """
+    def __init__(self, inputs):
+        # Combine args and kwargs:
+        self._has_variadic_arguments = True
+        self._varnames = list(inputs.keys())
+        self._defaults = [inspect.Parameter.empty for _ in self._varnames]
+        self._warned_not_contiguous_input = False
+
+
+# A customized None type for ComplexDataManager
+ComplexDataManagerNone = object()
+
+
+class ComplexDataManager:
```

Comment:
It would be helpful if an example is provided to show how input is converted to the target data structure.